experimental/yarpl/include/reactivestreams/ReactiveStreams.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <exception>
+#include <memory>
+
+namespace reactivestreams_yarpl {
+// TODO remove 'yarpl' if we agree on this usage elsewhere
+
+class Subscription {
+ public:
+  virtual ~Subscription() = default;
+  virtual void request(int64_t n) = 0;
+  virtual void cancel() = 0;
+};
+
+template <typename T>
+class Subscriber {
+ public:
+  virtual ~Subscriber() = default;
+  virtual void onSubscribe(Subscription*) = 0;
+  virtual void onNext(const T&) = 0;
+  virtual void onNext(T&& value) {
+    onNext(value);
+  }
+  virtual void onComplete() = 0;
+  virtual void onError(const std::exception_ptr error) {
```

Comment:
I guess you are trying to avoid dependency to folly.
Is there an equivalent in boost?